Add logic_bat_ball_problem

opendining opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, __failure to follow the guidelines below will result in the PR being closed automatically__. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access granted. 🚨

__PLEASE READ THIS__:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

Also, pelase note that we're using **Git LFS** for storing the JSON files, so please make sure that you move the JSON file to Git LFS before submitting a PR. Details on how to use Git LFS are available [here](https://git-lfs.com).

## Eval details 📑
### Eval name
Logic bat ball problem

### Eval description

The bat/ball problem is a classic math puzzle, usually expressed as:
A bat and a ball cost $1.10 in total. The bat costs $1 more than the ball. How much does the ball cost?
I found that GPT-3.5 and GPT-4 gave incorrect answers as the problem varied.

### What makes this a useful eval?

Logical reasoning is important, and the model is generally inaccurate with this dataset.

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can  create an eval on cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [x] **Include at least 15 high quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

3.5 failed almost all the time (0.0038 accuracy)
GPT-4 failed in my spot checks also

## Eval structure 🏗️

Your eval should
- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your yaml is registered at `evals/registry/evals/{name}.yaml`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (https://platform.openai.com/docs/usage-policies).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgement

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access granted.

### Submit eval

- [x] I have filled out all required fields of this form
- [x] I have used **Git LFS** for the Eval JSON data
- [x] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data 

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
{"input":[{"role":"system","content":"A bat and a ball cost $1.10 in total. The bat costs $1 more than the ball. How much does the ball cost? Answer in number only, with no explanation, and with no formatting or dollar signs."}],"ideal":"0.05"}
{"input":[{"role":"system","content":"A bat and a ball cost $1.20 in total. The bat costs $1 more than the ball. How much does the ball cost? Answer in number only, with no explanation, and with no formatting or dollar signs."}],"ideal":"0.10"}
{"input":[{"role":"system","content":"A bat and a ball cost $1.30 in total. The bat costs $1 more than the ball. How much does the ball cost? Answer in number only, with no explanation, and with no formatting or dollar signs."}],"ideal":"0.15"}
{"input":[{"role":"system","content":"A bat and a ball cost $1.40 in total. The bat costs $1 more than the ball. How much does the ball cost? Answer in number only, with no explanation, and with no formatting or dollar signs."}],"ideal":"0.20"}
{"input":[{"role":"system","content":"A bat and a ball cost $1.50 in total. The bat costs $1 more than the ball. How much does the ball cost? Answer in number only, with no explanation, and with no formatting or dollar signs."}],"ideal":"0.25"}
{"input":[{"role":"system","content":"A bat and a ball cost $1.60 in total. The bat costs $1 more than the ball. How much does the ball cost? Answer in number only, with no explanation, and with no formatting or dollar signs."}],"ideal":"0.30"}
{"input":[{"role":"system","content":"A bat and a ball cost $1.70 in total. The bat costs $1 more than the ball. How much does the ball cost? Answer in number only, with no explanation, and with no formatting or dollar signs."}],"ideal":"0.35"}
  ```
</details>

dyar-al-ashtari commented:
Both models answer correctly if asked to explain the answer, but both indeed fail if asked to answer in one number. Very interesting!

opendining commented:
> Both models answer correctly if asked to explain the answer, but both indeed fail if asked to answer in one number. Very interesting!

Yeah, I found that until I added the "with no explanation" bit, 3.5 would always elaborate, which isn't the intent. 4 would skip the elaboration, but still fail to answer correctly.

usama-openai reviewed (changes requested):
Thanks for the contribution. This PR looks interesting. However, I would like to request some changes.

1. More than `3900+` samples are too many. Add maximum `2000` samples in dataset.

2. It seems like you've used a generator script to create the dataset. It would be much appreciated if you placed the sample generation script in the `evals/registry/data/logic_bat_ball_problem/` directory.

We would love to review the PR again after the suggested changes.

opendining commented:
@usama-openai Thanks for checking this! I've updated the dataset to include only 385 samples, it still has 0.010 accuracy against 3.5-turbo. I also added a generator script for it.

usama-openai reviewed (approved):
This PR looks in good shape now. I'm approving this PR.

usama-openai reviewed (changes requested):
Thanks for the contribution. I would like to request a change.

I would recommend asking the model to reason before answering and using the `Includes` evaluation method here because arithmetic and other complex tasks are hard for the model to do zero-shot without a chance to reason through the steps. Asking the model to provide reasoning will give the model a fair chance to solve the question, and a proper evaluation method can be used to evaluate the provided answer. You can also add instructions to provide output in a specific format and then use that format to write the ideal answer. For example, the model can be asked to enclose the final answer in square brackets, and the ideal answer can be formatted like `[5]`. It'll give the model a proper chance to reason before answering, and the proper formatting will help in identifying the final answer using the `Includes` method.

usama-openai commented:
Closing the PR due to inactivity; please feel free to reopen if you get a chance to address comments.
